docs/proposals/compute-resource-metrics-api.md

```diff
@@ -0,0 +1,195 @@
+# Kubernetes compute resource metrics API
+
+## Goals
+
+Provide resource usage metrics on pods and nodes on the API server to be used
+by the scheduler to improve job placement, utilization, etc. and by end users
+to understand the resource utilization of their jobs.
+
+## Current state
+
+Right now, the Kubelet exports container metrics via an API endpoint. This
+information is not gathered nor served by the Kubernetes API server.
+
+## Use cases
+
+The first user will be kubectl. The resource usage data can be shown to the
+user via a periodically refreshing interface similar to `top` on Unix-like
+systems. This info could let users assign resource limits more efficiently.
+
+```
+$ kubectl top kubernetes-minion-abcd
```

Comment:
+1. That would be useful. 

`kubectl get` should work on these API resources, like like all other API resources:
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/kubectl/cmd/get.go

One wrinkle is that these are the first subresources we want to display. Probably the syntax should be something like `podmetrics/derived`.

It would be useful to add a general option (--sort=column, -scolumn) to request to sort by particular columns of `get` output. `top` could be an alias for `get podmetrics/derived -w -s2 -s3`.